src/components/export.tsx

```diff
@@ -1,51 +1,221 @@
+  }, [reference]);
 
+  const startExporting = (option: any) => () => {
+    const formatType = option.type;
 
-  const renderExportIcon = () => {
-    if (showSpinner) return (<span className='chaise-btn-icon fa-solid fa-rotate fa-spin' />);
+    switch(formatType) {
+      case 'DIRECT':
+        location.href = reference.csvDownloadLink;
+        break;
+      case 'BAG':
+      case 'FILE': 
+        setShow(true);
+        setSelectedOption(option);
+        const bagName = reference.table.name;
+        const exporter = new ConfigService.ERMrest.Exporter(
+          reference,
+          bagName,
+          option,
+          ConfigService.chaiseConfig.exportServicePath
+        );
+        setExporterObj(exporter);
+        if (exporter) {
+          const logStack = LogService.addExtraInfoToStack(null, {
+              template: {
+                  displayname: exporter.template.displayname,
+                  type: exporter.template.type
+              }
+          });
+          const logObj = {
+              action: LogService.getActionString(LogActions.EXPORT),
+              stack: logStack
+          }
+          exporter
+            .run(logObj)
+            .then((response: any) => {
+              // if it was canceled, just ignore the result
+              setShow(false);
+              if (response.canceled) return;
+              location.href = response.data[0];
+            })
+            .catch((error: any) => {
+              setShow(false);
+            });
+        }
+        break;
+    }
+  };
 
-    return <span className='chaise-btn-icon fa-solid fa-file-export' />;
+  const cancelExport = () => {
+    // Cancel download
+    if (!!exporterObj) {
+      exporterObj.cancel();
+    }
   }
 
+  const closeModal = () => {
+    cancelExport();
+    setShow(false);
+
+    // Call parent component callback to show alert box
+    onCancelDownload();
+  };
+
+  /**
+   * returns Modal Component - Component that renders progress modal dialog
+   */
+  const ExportModal = ({ title, show, closeModal }: ExportModalProps) => {
+    return (
+      <Modal
+        show={show}
+        onHide={closeModal}
+        backdrop='static'
+        keyboard={false}
+      >
+        <Modal.Header>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className='modal-text'>Your request is being processed...</div>
+          <div className='modal-text'>
+            You will be prompted to download the file when it is ready.
+          </div>
+          <div className='progress'>
+            <div
+              className='progress-bar progress-bar-striped active'
+              aria-valuenow={100}
+              aria-valuemax={100}
+              style={{ width: '100%' }}
+            ></div>
+          </div>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button className='chaise-btn chaise-btn-primary' variant='outline-primary' onClick={closeModal}>
```

Comment:
Use the `chaise-btn chaise-btn-secondary` class for this button (the same as AngularJS version).